examples/EntityAnimation/main.cpp

```diff
@@ -81,14 +85,21 @@ int main( int argc, char* argv[] ) {
     //! [Cache the camera manager]
 
     //! [Add usefull custom key events]
-    auto callback = [cameraManager]( QKeyEvent* event ) {
-        // Convert ascii code to camera index
-        cameraManager->activate( event->key() - '0' );
+    auto callback = [cameraManager]( QEvent* event ) {
+        if ( event->type() == QEvent::KeyPress ) {
+            auto keyEvent = static_cast<QKeyEvent*>( event );
+            // Convert ascii code to camera index
+            cameraManager->activate( keyEvent->key() - '0' );
+        }
     };
-    app.m_mainWindow->getViewer()->addKeyPressEventAction(
-        "switchCam0", "Key_0", "ShiftModifier", "", "false", callback );
-    app.m_mainWindow->getViewer()->addKeyPressEventAction(
-        "switchCam1", "Key_1", "ShiftModifier", "", "false", callback );
+    app.m_mainWindow->getViewer()->addCustomAction(
+        "switchCam0",
+        Gui::KeyMappingManager::createEventBindingFromStrings( "", "ControlModifier", "Key_0" ),
```

Comment:
Key event are managed on the Qt side, on US tenkeyless keyboard it's not possible to handle Shift+Key_0 (which triggers shift modifier and key ')') for instance. So yes possible key combination do depends on the keyboard layout.
That's all the purpose of custom key mapping so that you can easily tweak key mapping to work with your layout.
For this demo, and for default config, we may remove any key combination that uses keypad, or key combined value (e.g. numbers, #$%@ and so on).

Reply:
Strange but is Qt stuff ... so, ok to prevent the use of modifier-based keys.